tsconfig.json

```diff
@@ -8,13 +8,19 @@
     "sourceMap": true,
     "declaration": true,
     "outDir": "./dist",
+    "lib": [
+      "es5",
+      "es2015.iterable",
+      "es2015.collection",
+      "es2015.promise",
+      "dom"
+    ],
```

Comment:
I had to remove es6-shim typings, because they weren't compatible with angular, they switched to core-js for the polyfills: https://github.com/angular/angular/pull/10884. 

They don't use extra typings for the Promise type, instead they take the ones from the es2015 library bundled in Typescript (https://www.typescriptlang.org/docs/handbook/compiler-options.html). 

I looked at the ones angular itself is using and included the same: https://github.com/angular/angular/blob/master/modules/tsconfig.json